library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -343,6 +346,7 @@ private void runInternal() {
      * handle the response, and transfer the data to the destination file.
      */
     private void executeDownload(State state) throws StopRequestException {
+        checkPausedOrCanceled(originalDownloadInfo);
```

Comment:
if the task is already submitted to the executor, it will start part of the download and only after check whether it should be paused